src/fennel/utils.fnl

```diff
@@ -316,6 +323,14 @@ Returns nil if passed something other than a multi-sym."
 (fn quoted? [symbol]
   symbol.quoted)
 
+(fn idempotent-expr? [x]
+  "Checks if an object is an idempotent expression. Returns the object if it is."
+  (or (= (type x) :string)
+      (= (type x) :integer)
+      (= (type x) :number)
+      (and (sym? x)
+           (not (multi-sym? x)))))
```

Comment:
This would be really useful to expose in macro scope. (The built-in macros basically implement their own version of this.) We can see about doing that after merging this PR.